### ISSUE TITLE
Support for OpenAI's code interpreter, retrieval, function tools

### ISSUE THREAD

**Issue by vinodvarma24** [xlang-ai/OpenAgents#80]
Since OpenAI DevDay released amazing updates yesterday, specifically the tools(code_interpreter, retrieval, functions), It would be great to have these tools available in Data Agent and Plugin Agent.

**Comment by Timothyxxx** [comment]
Yes!! I think OpenAI has specialized models optimized for better performance in doing these tasks. These can be good tools or even parts of the foundation insides the framework. Do you have some comments on that?

**Comment by vinodvarma24** [comment]
Yes, along with the new model(gpt-4-1106-preview), I think we should have the openai code_interpreter tool as the default python execution sandbox environment as a first improvement. Then we can add the retrieval, function tools.
